main.py

```diff
@@ -1,38 +1,85 @@
 import torch
 
 from models.SpatioTemporalTransformer import SpatioTemporalTransformer
+# from models.BiDrectionalTransformer import BiDirectionalTransformer
 from utils.dataset import prepare_dataset, generate_auto_regressive_targets
 from utils.types import ModelEnum, TargetEnum
+# from utils.maskgenerator import BatchJointMaskGenerator
+import argparse
 
-
-def main(model_name: str = None, target_type: str = None) -> None:
+def main() -> None:
     """
     Main runner.
 
     Args:
-        model_name: The name of the model (used to select which model to run)
-        target_type: The type of target (used to determine post-processing logic)
+        None
 
     Returns:
         None
     """
+
+    # Process command line arguments
+    parser = argparse.ArgumentParser(prog="Model Training",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+                                     description="Train selected model with given parameters on chosen dataset.")
+    
+    model_info = parser.add_argument_group("Model Information")
+    model_info.add_argument("--model",choices =["STT","RNN","RNN_a","LSTM","LSTM_a","S2S","BDT"],default="STT")
+    model_info.add_argument("-tt","--target_type",choices=["default","auto","pretrain"],default="default")
+    model_info.add_argument("-dp","--data_path",default="data/sampled/aa/")
+
+    parameters = parser.add_argument_group("Model Parameters")
+    parameters.add_argument("-b","--batch_size",default=64)
+    parameters.add_argument("-emb","--embedding_dim",default=64)
+    parameters.add_argument("-ep","--epochs",default=10)
+    parameters.add_argument("-h","--nhead",default=12)
+    parameters.add_argument("-enc","--encoder_layers",default=6)
+    parameters.add_argument("-f","--feedforward_dim",default=1024)
+
+    args = parser.parse_args()
+
     # Process train, val, test datasets
-    fpath = "data/sampled/aa/"
-    batch_size = 64
+    fpath = args.data_path
+    batch_size = args.batch_size
     device = 'cpu'
     datasets = prepare_dataset(fpath, batch_size, device)
 
     # Setup Model
-    _, seq_len, raw_dim = next(iter(datasets["train"]))[0].shape
-    embedding_dim = 64  # Size of embeddings
+    _, seq_len, raw_dim = next(iter(datasets["train"]))[1].shape
```

Comment:
Can you change this back to `_, seq_len, raw_dim = next(iter(datasets["train"]))[0].shape`?